### DISCUSSION TITLE
Optiscaler working without nvgx.dll

### DISCUSSION THREAD

**Discussion by Vaashx** [optiscaler/OptiScaler#90] [Q&A]
So, in the near future, are we gonna be able to get the optiscaler to work without need to get file DLSS or any file like that?, but directly having upscaler and Frame gen incorporated in the optiscaler, like having a game that doesn't support DLSS or FSR and be able to have frame gen and upscaler from optiscaler and not by the nvgx.dll or anything else from the side to be able to get those. Or it is impossible get it working ?

**Comment by cdozdil** [comment]
Hi,

If I understand correctly you want OptiScaler to support games does not have support for any kind of upscaling technology (FSR/XeSS/DLSS). This means per game effort like PureDark's mods. A general solution TAA like OptiScaler is not possible for these games is not possible but you can use spatial upscalers like [Magpie](https://github.com/Blinue/Magpie) for this purpose.

**Reply by Vaashx** [reply]
i already running with lossless scaling for this, but yes and no, it's hard to explain, it's like having those techno ( FSR/DLSS/XeSS ) directly in the DXGI api and being able to use them on game that doesn't support any kind of upscaler, all the upscaling thing worked trought the optiscaler, i don't know how to explain this, see that like using Reshade but instead of using color settings it will be replaced by upscalers, reflex/antilag2 and frame gen. I know maybe literally impossible to do, or maybe too complicated to make it work, i don't know, i would love to see this becoming real.
